Assignments/ip2/ip2.md

```diff
@@ -0,0 +1,385 @@
+* Test PosterSessionAreaController imageContents property: 7 points
+* Test PosterSessionAreaController stars property: 7 points
+* Test PosterSessionAreaController toPosterSessionAreaModel: 2 points
+
+Partial marks are available for detecting some (but not all) faults. The number of faults detected may not directly correlate with the difficulty of writing the test: there are several faults that are nearly guaranteed to be detected together (writing a test that finds one of them is guaranteed to find both of them), which is why there are different cutoffs for partial and full marks for the tests.
+
+### Task 2: Implement React Hooks (50 points)
+As discussed in Module 8, an effective pattern for building React applications is to use hooks within components to access global state. As part of the refactoring to implement the Interactable abstraction throughout Covey.Town, Avery also refactored the entire React-based frontend to use this pattern of hooks. Before implementing the final component that displays and synchronizes video playback in Viewing Areas, your next task will be to implement these hooks - some of which are related to the interactables, and some of which are related to Avery’s overall refactoring to use more hooks.
+
+Some of these hooks may require you to include `useEffect` and/or `useState` hooks within the hook that you are building. For each of the hooks, consider the events that they might need to listen to.
+The hooks you'll be implementing are in `PosterSessionAreaController`.
+
+Be sure to follow the [rules of hooks](https://reactjs.org/docs/hooks-rules.html) when implementing your hooks - these will be enforced by the linter, and also by the TAs when grading for style.
+
+{::options parse_block_html="true" /}
+<details>
+<summary markdown="span">View the specification for these tasks</summary>
+
+
+
+PosterSessionAreaController
+
+{% highlight typescript %}
+/**
+ * A hook that returns the number of stars for the poster session area with the given controller
+ */
+export function useStars(controller: PosterSessionAreaController): number
+
+/**
+ * A hook that returns the image contents for the poster session area with the given controller
+ */
+export function useImageContents(controller: PosterSessionAreaController): string | undefined
+
+/**
+ * A hook that returns the title for the poster session area with the given controller
+ */
+export function useTitle(controller: PosterSessionAreaController)
+{% endhighlight %}
+</details>
+
+Grading for Task 2:
+You do not need to write any tests for task 2. The handout contains all of the tests that our grading script will use.
+To run the tests, run `npm test PosterHooks` in the `frontend` directory.
+
+Point break down for each of the implementation tasks:
+* Implement PosterSessionAreaController.ts useStars: 10 points
+* Implement PosterSessionAreaController.ts useImageContents: 10 points
+* Implement PosterSessionAreaController.ts useTitle: 10 points
```

Comment:
I think we just need these to add to 50. I double checked the handout and it looks good for this section too!

Reply:
wow good catch